### PR TITLE
added button-card

### PR DESCRIPTION
**Pull request by gauravsinhaweb**
Added new components i.e. button, card

### CODE REVIEW

**Review by PuneetP16 — APPROVED** [review]
Awesome look!

File: components/alert/alert.css [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
-  background-color: #f6f6f6;
-  color: #616161;
+}
+.alert-primary {
+  background-color: var(--primary-li);
+  color: var(--primary);
+  border-left: 4px solid var(--primary);
+}
+.alert-secondary {
+  background-color: #ebebeb;
```
Comment (PuneetP16) [inline components/alert/alert.css]: Use variables instead of direct values in the case of colours.

File: components/button/button.css [hunk 2]
```diff
@@ -0,0 +1,113 @@
+.btn {
+  border-radius: 5px;
+  color: var(--white);
+  padding: 1rem 1.5rem;
+  font-size: 1.5rem;
+  font-weight: 500;
+  transition: all ease 0.5s;
+  text-transform: capitalize;
+  box-shadow: 0 3px 1px -2px #0003, 0 2px 2px #00000024, 0 1px 5px #0000001f;
```
Comment (PuneetP16) [inline components/button/button.css:9]: add colour variable.

File: components/button/button.html [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    <div class="container dark_theme">
+      <!-- Navbar  -->
+      <nav class="wrapper_nav flex">
+        <h1>
+          <a href="../../index.html">
+            <span title="pixel-ui" class="h2 nav_brand_text"> PIXEL</span></a
+          >
+          <!-- <span>
+            <img src="./assets/splash.svg" alt="logo" class="nav_brand" />
+          </span> -->
```
Comment (PuneetP16) [inline components/button/button.html:23]: Remove commented lines.

**Review by PuneetP16 — APPROVED** [review]

File: components/button/button.css [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  transition: all 400ms cubic-bezier(0.25, 0.8, 0.25, 1),
+    box-shadow 280ms cubic-bezier(0.4, 0, 0.2, 1);
+  cursor: pointer;
+}
+.btn:active {
+  box-shadow: 0 5px 5px -3px #0003, 0 8px 10px 1px #00000024,
+    0 3px 14px 2px #0000001f;
+}
+.btn:disabled,
+.btn[disabled] {
```
Comment (Shivani-1524) [inline components/button/button.css:19]: just a suggestion for disabled button to not have cursor pointer.

File: components/button/button.css [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  font-size: 2rem;
+  padding: 1rem 1.5rem;
+  cursor: pointer;
+  text-decoration: underline;
+  opacity: 0.8;
+}
+.btn-link:hover {
+  opacity: 1;
+}
+.out-primary {
```
Comment (Shivani-1524) [inline components/button/button.css:78]: maybe add a hover effect for the outlined buttons

File: components/alert/alert.css [hunk 6]
```diff
@@ -1,25 +1,37 @@
 .alert {
   border: 0.5px solid;
```
Comment (bhushannaikaroskar) [inline components/alert/alert.css]: I think 1 px is the least possible unit please use integers not floating point number when defining px.

**Review by bhushannaikaroskar — CHANGES_REQUESTED** [review]

File: components/button/button.css (hunk 4 above)
Comment (gauravsinhaweb) [reply components/button/button.css:19]: yeah! I'll do that

File: components/alert/alert.css (hunk 6 above)
Comment (gauravsinhaweb) [reply components/alert/alert.css]: sure, thanks for pointing it out.

File: components/button/button.css (hunk 4 above)
Comment (gauravsinhaweb) [reply components/button/button.css:19]: > just a suggestion for disabled button to not have cursor pointer.

yeah! I'll do that .

File: components/button/button.css (hunk 5 above)
Comment (gauravsinhaweb) [reply components/button/button.css:78]: > maybe add a hover effect for the outlined buttons
that's intentional actually, will improve the hover effect

File: components/button/button.html (hunk 3 above)
Comment (gauravsinhaweb) [reply components/button/button.html:23]: > Remove commented lines.

yeah, I'll do that

**Comment by gauravsinhaweb** [comment]
> Awesome look!

Thanks!!

File: components/button/button.css (hunk 4 above)
Comment (nikhiljugale007) [reply components/button/button.css:19]: You can use this property. cursor: no-drop

**Comment by nikhiljugale007** [comment]
The card looks good.
In the button, you change the background color of secondary colors on hover. Other than that everything looks godd.

File: components/badge/badge.html [hunk 7]
```diff
@@ -3,66 +3,221 @@
...
+          </span> -->
+        </h1>
+
+        <div class="wrapper_nav_menu">
+          <ul class="nav_menu flex">
+            <li class="p">
+              <a
+                href="https://twitter.com/gauravsinhaweb"
+                rel="noopener noreferrer"
+                target="_blank"
```
Comment (iRohitGaur) [inline components/badge/badge.html]: Setting target="_blank" on <a> elements now implicitly provides the same rel behavior as setting rel="noopener"
Read:
https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noopener

File: components/button/button.css [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+}
+.btn:active {
+  box-shadow: 0 5px 5px -3px #0003, 0 8px 10px 1px #00000024,
+    0 3px 14px 2px #0000001f;
+}
+.btn:disabled,
+.btn[disabled] {
+  border: 1px solid #999999;
+  background-color: #cccccc;
+  color: #666666;
```
Comment (iRohitGaur) [inline components/button/button.css]: use variables

File: components/card/card.css [hunk 9]
```diff
@@ -0,0 +1,162 @@
+/* basic card  */
+
+.card {
+  background-color: #fff;
+
+  overflow: hidden;
+  object-fit: fill;
+  width: 35rem;
+  border: 1px solid #ccc;
+  border-radius: 0.5rem;
```
Comment (iRohitGaur) [inline components/card/card.css]: use px for values less than 10px

File: components/input/input.html [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+      <!-- Navbar  -->
+      <nav class="wrapper_nav flex">
+        <h1>
+          <a href="../../index.html">
+            <span title="pixel-ui" class="h2 nav_brand_text"> PIXEL</span></a
+          >
+
+          <!-- <span>
+            <img src="./assets/splash.svg" alt="logo" class="nav_brand" />
+          </span> -->
```
Comment (iRohitGaur) [inline components/input/input.html:24]: remove commented code

File: core.css [hunk 11]
```diff
@@ -109,5 +140,57 @@ p {
...
+/* .toolbar {
+  position: sticky !important;
+} */
+.copy-to-clipboard-button,
+.toolbar,
+.toolbar-item {
+  background: 0 0;
+  border: 0;
+  font: inherit;
+  font-size: 1.6rem !important;
```
Comment (iRohitGaur) [inline core.css:196]: avoid using important

**Review by iRohitGaur — CHANGES_REQUESTED** [review]
Good to merge with changes 👍🏻

File: components/badge/badge.html [hunk 12]
```diff
@@ -3,66 +3,221 @@
...
+                    <path
+                      fill-rule="evenodd"
+                      d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59c.4.07.55-.17.55-.38c0-.19-.01-.82-.01-1.49c-2.01.37-2.53-.49-2.69-.94c-.09-.23-.48-.94-.82-1.13c-.28-.15-.68-.52-.01-.53c.63-.01 1.08.58 1.23.82c.72 1.21 1.87.87 2.33.66c.07-.52.28-.87.51-1.07c-1.78-.2-3.64-.89-3.64-3.95c0-.87.31-1.59.82-2.15c-.08-.2-.36-1.02.08-2.12c0 0 .67-.21 2.2.82c.64-.18 1.32-.27 2-.27c.68 0 1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82c.44 1.1.16 1.92.08 2.12c.51.56.82 1.27.82 2.15c0 3.07-1.87 3.75-3.65 3.95c.29.25.54.73.54 1.48c0 1.07-.01 1.93-.01 2.2c0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
+                      fill="currentColor"
+                    ></path>
+                  </svg>
+                </div>
+              </a>
+            </li>
+            <li class="p">
```
Comment (mehulsatardekar) [inline components/badge/badge.html:76]: the naming of class name can be improved here not sure what is the purpose of .p class here

File: components/card/card.css [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+  justify-content: flex-end;
+  align-items: left;
+  text-align: left;
+  margin-bottom: 2.5rem;
+  padding: 0 1rem;
+  color: #fff;
+  filter: grayscale(100%);
+}
+.card-content {
+  padding: 16px;
```
Comment (mehulsatardekar) [inline components/card/card.css]: use rem here

**Review by mehulsatardekar — CHANGES_REQUESTED** [review]

**Comment by mehulsatardekar** [comment]
awesome job, the UI looks clean and neat. 👌

File: components/card/card.css (hunk 9 above)
Comment (gauravsinhaweb) [reply components/card/card.css]: sure! thanks for pointing it out

File: components/card/card.css (hunk 13 above)
Comment (gauravsinhaweb) [reply components/card/card.css]: sure! thanks for pointing it out.

**Comment by gauravsinhaweb** [comment]
> awesome job, the UI looks clean and neat. 👌

Thanks, Mehul!!

File: components/badge/badge.html (hunk 12 above)
Comment (gauravsinhaweb) [reply components/badge/badge.html:76]: > the naming of class name can be improved here not sure what is the purpose of .p class here

I've made my own typography style. so "p" stands for paragraph.

File: core.css (hunk 11 above)
Comment (gauravsinhaweb) [reply core.css:196]: > avoid using important

I'm overriding prismjs CSS, without !important, it won't work. let me know if you have a solution for that.

File: components/button/button.css (hunk 8 above)
Comment (gauravsinhaweb) [reply components/button/button.css]: > use variables

yeah, i will do that.

File: components/badge/badge.html (hunk 7 above)
Comment (gauravsinhaweb) [reply components/badge/badge.html]: > Setting target="_blank" on elements now implicitly provides the same rel behavior as setting rel="noopener" Read: https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noopener

thanks for sharing this, I will improve it.

**Comment by gauravsinhaweb** [comment]
> The card looks good. In the button, you change the background color of secondary colors on hover. Other than that everything looks godd.

yeah, I'll improve the hover effect. Thanks for the compliment :)

File: components/button/button.css (hunk 4 above)
Comment (gauravsinhaweb) [reply components/button/button.css:19]: I did that, thanks!

**Pull request merged**